pkg/test/resourcefixture/contexts/register.go

```diff
@@ -452,16 +452,19 @@ func directDelete(ctx context.Context, u *unstructured.Unstructured, c client.Cl
 	// in subsequent calls but found it originally, we can consider it deleted.
 	firstFind := false
 
+	var deleteError error
 	for i := 0; i < maxRetries; i++ {
 		found, err := a.Find(ctx)
-		if err == nil && found {
+		deleteError = err
+
+		if deleteError == nil && found {
```

Comment:
Keep this one as `if err == nil && found {`, it makes the logic easier to follow (IMO)